src/output-model.js

```diff
@@ -0,0 +1,93 @@
+
+import {EventEmitter} from 'events';
+
+/**
+ * Provides Jupyter outputs to an output area.
+ *
+ * This class manages the conversion of Jupyter messages to output state, and
+ * handles adding the output state to the output area.
+ */
+export class OutputModel extends EventEmitter {
+    
+    
+    /**
+     * Public constructor
+     */
+    constructor() {
+        super();
+        this.clearQueued = false;
+        this._state = [];        
```

Comment:
I was looking at renaming the getter/setter to `setState` and `getState`, and doing the closure thing. I changed my mind last minute, no reason other than the appreciation of the get and set keywords.